src/api/app/models/relationship.rb

```diff
@@ -125,6 +127,12 @@ def self.with_groups_and_roles
     with_groups_and_roles_query.pluck('groups.title', 'roles.title')
   end
 
+  def event_delete
+    return if User.session == user
```

Comment:
Could we rely on Pundit here instead? This way, authorization code is only in policies. It's obviously not the case right now in OBS, but it's the end goal.

Reply:
Oh alright!